biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -471,15 +480,31 @@ private int activate() throws Exception {
 		return result;
 	}
 
-	private boolean isImmediate(BundleActivator activator) {
+	private Object getImmediateFieldValue(BundleActivator activator) {
 		try {
 			Field f = activator.getClass()
 				.getField("IMMEDIATE");
+			return f.get(activator);
+		} catch (Exception e) {}
+		return null;
+	}
 
-			return f.getBoolean(activator);
-		} catch (Exception e) {
-			return false;
+	private boolean isAfterFrameworkInit(BundleActivator activator) {
+		Object result = getImmediateFieldValue(activator);
+		return "AFTER_FRAMEWORK_INIT".equals(result);
+	}
+
+	private boolean isBeforeBundleStart(BundleActivator activator) {
```

Comment:
There is a plurality mismatch between the method name (Bundle singular) and the constant value name (BUNDLES plural). They should match.